File: src/main/java/net/frozenblock/mrbeast/entity/render/renderer/MrBeastRenderer.java
package net.frozenblock.mrbeast.entity.render.renderer;

import net.frozenblock.mrbeast.entity.MrBeast;
import net.frozenblock.mrbeast.entity.render.model.MrBeastModel;
import net.frozenblock.mrbeast.util.MrBeastSharedConstants;
import net.minecraft.client.model.HumanoidArmorModel;
import net.minecraft.client.model.geom.EntityModelSet;
import net.minecraft.client.model.geom.ModelLayerLocation;
import net.minecraft.client.model.geom.ModelLayers;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import net.minecraft.client.renderer.entity.HumanoidMobRenderer;
import net.minecraft.client.renderer.entity.layers.HumanoidArmorLayer;
import net.minecraft.resources.ResourceLocation;

public class MrBeastRenderer extends HumanoidMobRenderer<MrBeast, MrBeastModel<MrBeast>> {
	public MrBeastRenderer(EntityRendererProvider.Context context, ModelLayerLocation mainLayer, ModelLayerLocation innerArmorLayer, ModelLayerLocation outerArmorLayer) {
		super(context, createModel(context.getModelSet(), mainLayer), 0.5F);
		this.addLayer(
				new HumanoidArmorLayer<>(
						this,
						new HumanoidArmorModel<>(context.bakeLayer(innerArmorLayer)),
						new HumanoidArmorModel<>(context.bakeLayer(outerArmorLayer)),
						context.getModelManager()
				)
		);
	}

	private static MrBeastModel<MrBeast> createModel(EntityModelSet modelLoader, ModelLayerLocation layer) {
		return new MrBeastModel<>(modelLoader.bakeLayer(layer));
	}

	@Override
	public ResourceLocation getTextureLocation(MrBeast entity) {
		return MrBeastSharedConstants.id("textures/entity/beast.png");
	}
}

File: src/main/java/net/frozenblock/mrbeast/entity/render/renderer/MrBeastRenderer.kt
package net.frozenblock.mrbeast.entity.render.renderer

import net.frozenblock.mrbeast.entity.MrBeast
import net.frozenblock.mrbeast.entity.render.model.MrBeastModel
import net.frozenblock.mrbeast.util.MrBeastSharedConstants.id
import net.minecraft.client.model.geom.EntityModelSet
import net.minecraft.client.model.geom.ModelLayerLocation
import net.minecraft.client.renderer.entity.EntityRendererProvider
import net.minecraft.client.renderer.entity.HumanoidMobRenderer
import net.minecraft.resources.Identifier

class MrBeastRenderer(
    context: EntityRendererProvider.Context,
    mainLayer: ModelLayerLocation,
    innerArmorLayer: ModelLayerLocation,
    outerArmorLayer: ModelLayerLocation
) : HumanoidMobRenderer<MrBeast, MrBeastRenderState, MrBeastModel>(context, createModel(context.modelSet, mainLayer), 0.5f) {
    init {
        // TODO 26.1
        /*addLayer(
            HumanoidArmorLayer<>(
                this,
                ArmorModelSet.bake(innerArmorLayer, context.modelSet, ::MrBeastModel)
                HumanoidArmorModel(context.bakeLayer(innerArmorLayer)),
                HumanoidArmorModel(context.bakeLayer(outerArmorLayer)),
                context.modelManager
            )
        )*/
    }

    override fun getTextureLocation(state: MrBeastRenderState): Identifier {
        return id("textures/entity/mrbeast.png")
    }

    override fun createRenderState(): MrBeastRenderState {
        return MrBeastRenderState()
    }

    companion object {
        private fun createModel(modelLoader: EntityModelSet, layer: ModelLayerLocation): MrBeastModel {
            return MrBeastModel(modelLoader.bakeLayer(layer))
        }
    }
}
